### PR TITLE
added pre-release workflow

### PR DESCRIPTION
**Pull request by gat786**
## Changes

- Added a pre-release workflow that can be called to generate Release Candidate releases

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or a feature that would cause existing functionality not to work as expected)
- [ ] Documentation update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional or API changes)
- [x] Build-related changes
- [ ] Other: _description_ 

#### Requires testing

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/workflows/pre-release-nethermind.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      id: buildx
+      uses: docker/setup-buildx-action@v2
+    - name: Log in to Docker Hub
+      if: success()
+      run: |
+        echo "${{ secrets.DOCKER_ACCESS_TOKEN }}" | docker login --username "${{ secrets.DOCKER_USERNAME }}" --password-stdin
+    - name: Build and push image to Docker registry (major) / trigger DAppNode build
+      if: ${{ !contains(github.event.inputs.tag, '-') }}
+      run: |
+        docker buildx build --platform=linux/amd64,linux/arm64 -t "${{ env.DOCKER_IMAGE }}:latest" -t "${{ env.DOCKER_IMAGE }}:${{ github.event.inputs.tag }}" -f Dockerfile --build-arg COMMIT_HASH=${{ needs.build.outputs.COMMIT_HASH }} --build-arg BUILD_TIMESTAMP=${{ needs.build.outputs.BUILD_TIMESTAMP }} . --push
```
Comment (FalcoXYZ) [inline .github/workflows/pre-release-nethermind.yml:143]: Remove latest from here and make it "unstable". 
You can leave the second one as this should push to the tag, which should be suffixed with RC1 for example.

File: .github/workflows/pre-release-nethermind.yml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        curl -s -X POST -u "${{ secrets.REPOSITORY_DISPATCH_TOKEN }}" -H "Accept: application/vnd.github.everest-preview+json" -H "Content-Type: application/json" -d '{"event_type":"dappnode","client_payload":{"tag":"${{ github.event.inputs.tag }}"}}' https://api.github.com/repos/$GITHUB_REPOSITORY/dispatches
```
Comment (FalcoXYZ) [inline .github/workflows/pre-release-nethermind.yml:144]: Let's remove this

File: .github/workflows/pre-release-nethermind.yml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    - name: Build and push image to Docker registry  (patch)
```
Comment (FalcoXYZ) [inline .github/workflows/pre-release-nethermind.yml:145]: Let's remove this as we are already pushing it above. (to "unstable" and to RC version number)

File: .github/workflows/pre-release-nethermind.yml [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+          ${{ github.workspace }}/${{ env.PACKAGE_DIR }}/
+          ${{ env.SCRIPTS_PATH }}/
+        key: packages-${{ needs.build.outputs.COMMIT_HASH }}
+    - name: Publish
+      env:
+        GIT_COMMIT: ${{ needs.build.outputs.COMMIT_HASH }}
+        GIT_TAG: ${{ github.event.inputs.tag }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        PACKAGE_PREFIX: ${{ needs.build.outputs.PACKAGE_PREFIX }}
+      run: $SCRIPTS_PATH/publish-github.sh
```
Comment (FalcoXYZ) [inline .github/workflows/pre-release-nethermind.yml:117]: You should probably update the github script and make it a pre-release or RC release. Not sure what's possible.

**Review by FalcoXYZ — COMMENTED** [review]
Left some comments for you.

**Comment by rubo** [comment]
Why not reuse the release workflow instead? It's hard to maintain multiple ones.

**Comment by kamilchodola** [comment]
@rubo 
We are pushing pivots right before releasing version so we would need to trigger same Release Pipeline twice with different settings.
It will be done later when Pivots problem will be solved - then we will be able to create better solution.
Also with this there is a better option to run multiple RC(even RC1 fails, we just trigger second one) and we are keeping Releases Action clear with only released version.

**Comment by rubo** [comment]
Put this on hold, guys. It's already done. Another PR is coming later today.

**Comment by rubo** [comment]
Closing in favor of #5086

**Pull request closed without merging**